utilities/datasets/read.py

```diff
@@ -61,7 +63,12 @@ def _parse_example(serialized_example):
     else:
         dataset.repeat()
```

Comment:
The way this is implemented will result in an infinite loop when running validation. Please fix it.

Reply:
Fixed. Nice catch!